controllers/datamover.go

```diff
@@ -628,6 +692,98 @@ func (r *DPAReconciler) checkIfDataMoverIsEnabled(dpa *oadpv1alpha1.DataProtecti
 	return false
 }
 
+func (r *DPAReconciler) checkDataMoverVolumeOptions(dpa *oadpv1alpha1.DataProtectionApplication) bool {
+
+	if dpa.Spec.Features != nil && dpa.Spec.Features.DataMover != nil &&
+		dpa.Spec.Features.DataMover.DataMoverVolumeOptions != nil {
+		return true
+	}
+
+	return false
+}
+
+func (r *DPAReconciler) buildDataMoverConfigMap(dpa *oadpv1alpha1.DataProtectionApplication, cm *corev1.ConfigMap) error {
+
+	if dpa == nil {
+		return fmt.Errorf("DPA CR cannot be nil")
+	}
+	if cm == nil {
+		return fmt.Errorf("datamover deployment cannot be nil")
+	}
+
+	cmMap := map[string]string{}
+
+	// check for source volume options
+	if dpa.Spec.Features.DataMover.DataMoverVolumeOptions.SourceVolumeOptions != nil {
+		sourceOptions := dpa.Spec.Features.DataMover.DataMoverVolumeOptions.SourceVolumeOptions
+
+		if len(sourceOptions.StorageClassName) > 0 {
+			cmMap["SourceStorageClassName"] = sourceOptions.StorageClassName
+		}
+
+		if len(sourceOptions.AccessMode) > 0 {
+			cmMap["SourceAccessMode"] = sourceOptions.AccessMode
+		}
+
+		if len(sourceOptions.CacheStorageClassName) > 0 {
+			cmMap["SourceCacheStorageClassName"] = sourceOptions.CacheStorageClassName
+		}
+
+		if len(sourceOptions.CacheAccessMode) > 0 {
+			cmMap["SourceCacheAccessMode"] = sourceOptions.CacheAccessMode
+		}
+
+		if len(sourceOptions.CacheCapacity) > 0 {
+			cmMap["SourceCacheCapacity"] = sourceOptions.CacheCapacity
+		}
+	}
+
+	// check for destination volume options
+	if dpa.Spec.Features.DataMover.DataMoverVolumeOptions.DestinationVolumeOptions != nil {
+		destinationOptions := dpa.Spec.Features.DataMover.DataMoverVolumeOptions.DestinationVolumeOptions
+
+		if len(destinationOptions.StorageClassName) > 0 {
+			cmMap["DestinationStorageClassName"] = destinationOptions.StorageClassName
+		}
+
+		if len(destinationOptions.AccessMode) > 0 {
+			cmMap["DestinationAccessMode"] = destinationOptions.AccessMode
+		}
+
+		if len(destinationOptions.CacheStorageClassName) > 0 {
+			cmMap["DestinationCacheStorageClassName"] = destinationOptions.CacheStorageClassName
+		}
+
+		if len(destinationOptions.CacheAccessMode) > 0 {
+			cmMap["DestinationCacheAccessMode"] = destinationOptions.CacheAccessMode
+		}
+
+		if len(destinationOptions.CacheCapacity) > 0 {
+			cmMap["DestinationCacheCapacity"] = destinationOptions.CacheCapacity
+		}
+	}
+
+	cm.Data = cmMap
+
+	return nil
+}
+
+func (r *DPAReconciler) checkConfigMapExists() (*corev1.ConfigMap, bool, error) {
+
+	// check configMap already exists
+	confmap := corev1.ConfigMap{}
+	err := r.Get(context.Background(), types.NamespacedName{Name: DataMoverConfigMapName, Namespace: r.NamespacedName.Namespace}, &confmap)
```

Comment:
I would make this function generic accepting any name/namespace.

As is, it can be a bit hard to know that DPAReconciler.checkConfigMapExists is specific to data mover without really digging.

Reply:
Just the name as parameter would be fine, we can reasonably assume anything coming out of `DPAReconciler` expect namespace to be `r.NamespacedName.Namespace`.

Reply:
Or rename the function to be more data mover specific